### ISSUE TITLE
nexus-inference: BnnF32 — binary neural network inference (I3)

### ISSUE THREAD

**Issue by Abso1ut3Zer0** [Abso1ut3Zer0/nexus#307]
## Summary

Binary neural network where weights are ±1. Matrix-vector multiply becomes XNOR + popcount — an order of magnitude faster than fp32 for the same architecture shape. First hidden layer remains fp32 (real-valued inputs), subsequent layers binary.

**Crate:** `nexus-inference`
**Module:** `nexus_inference::bnn::BnnF32`

## API sketch

```rust
pub struct BnnF32<const IN: usize, const H: usize, const OUT: usize> {
    w1_float: [[f32; IN]; H],      // first layer: real → binary
    b1: [f32; H],
    w2_packed: [[u64; H/64]; H],   // packed ±1 bits
    b2: [f32; H],
    w_out: [[f32; H]; OUT],        // output layer: binary → real
    b_out: [f32; OUT],
}
```

## Latency targets

~50 ns per binary hidden layer (H=64). ~100 ns for H=128.

## Design constraints

- Binary weights packed into u64 (one bit per weight)
- Activation is sign function (positive → 1, negative → 0 as bit)
- XNOR for multiply, popcount for accumulate
- First and last layers remain fp32 (real-valued I/O)
- Batch normalization folded into thresholds at load time
- `no_std` compatible

## Build order

**Second wave** — distinctive low-latency NN option.

*From nexus-inference roadmap item I3.*
